msgs/txDecoder.go

```diff
@@ -8,7 +8,11 @@ import (
 func TxDecoder(txBytes []byte) (sdk.Tx, sdk.Error) {
 	var spendMsg SpendMsg
 	if err := rlp.DecodeBytes(txBytes, &spendMsg); err != nil {
-		return SpendMsg{}, sdk.ErrTxDecode(err.Error())
+		var depositMsg IncludeDepositMsg
+		if err2 := rlp.DecodeBytes(txBytes, &depositMsg); err2 != nil {
+			return SpendMsg{}, sdk.ErrTxDecode(err.Error())
```

Comment:
Maybe include both errs in the return? Instead of `err.Error()` do something like `fmt.Sprintf("Decode to SpendMsg error: { %s } Decode to DepositMsg error: { %s }", err.Error(), err2.Error())`